### PR TITLE
[kube-state-metrics] Version bump to 2.0.0

### PR DESCRIPTION
**Pull request by mrueg**
#### What this PR does / why we need it:
* Version bump kube-state-metrics to 2.0.0
* Upgrade chart api version to v2

@tariq1890 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@bismarck @gianrubio @gkarthiks @scottrigby @vsliouniaev @Xtigyr can anyone take a look?

**Review by monotek — DISMISSED** [review]

File: charts/kube-state-metrics/templates/deployment.yaml [hunk 1]
```diff
@@ -69,91 +69,91 @@ spec:
...
-        - --collectors=verticalpodautoscalers
+        - --resources=verticalpodautoscalers
 {{  end  }}
 {{  if .Values.collectors.volumeattachments  }}
-        - --collectors=volumeattachments
+        - --resources=volumeattachments
 {{  end  }}
 {{ if .Values.namespace }}
-        - --namespace={{ .Values.namespace | join "," }}
+        - --namespaces={{ .Values.namespace | join "," }}
```
Comment (tariq1890) [inline charts/kube-state-metrics/templates/deployment.yaml:156]: Can we update the helm values `.Values.namespace` to `.Values.namespaces` to reflect this change?

Comment (tariq1890) [reply charts/kube-state-metrics/templates/deployment.yaml:156]: in L155 I mean

Comment (mrueg) [reply charts/kube-state-metrics/templates/deployment.yaml:156]: Sure we can. I'll add a migration guide then.

Comment (tariq1890) [reply charts/kube-state-metrics/templates/deployment.yaml:156]: Yes! Thanks a lot :)

Comment (mrueg) [reply charts/kube-state-metrics/templates/deployment.yaml]: Added a small upgrade guide and renamed this key.

**Review by monotek — APPROVED** [review]

**Pull request merged**